Sources/PackageDescription/Product.swift

```diff
@@ -8,46 +8,125 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-public struct Product {
-    public let name: String
-    public let type: ProductType
-    public let modules: [String]
+/// Defines a product in the package.
+public enum Product {
 
-    public init(name: String, type: ProductType, modules: String...) {
-        self.init(name: name, type: type, modules: modules)
+    /// An exectuable product.
+    public struct ExecutableProduct {
+
+        /// The name of the executable product.
+        public let name: String
+
+        /// The targets in the product.
```

Comment:
This comment should probably clarify that the array contains the _names_ of the targets.